pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -527,16 +533,20 @@ func getPodCIDRsOnNode(node *corev1.Node) []string {
 // createIPSecTunnelPort creates an IPSec tunnel port for the remote Node if the
 // tunnel does not exist, and returns the ofport number.
 func (c *Controller) createIPSecTunnelPort(nodeName string, nodeIP net.IP) (int32, error) {
+	portName := util.GenerateNodeTunnelInterfaceName(nodeName)
 	interfaceConfig, ok := c.interfaceStore.GetNodeTunnelInterface(nodeName)
+	// check if Node IP, PSK, or tunnel type changes. This can
+	// happen if removeStaleTunnelPorts fails to remove a "stale"
+	// tunnel port for which the configuration has changed, return error to requeue the Node.
```

Comment:
Since you removed the TODO, and according to the commit message, I suppose this is fixed. However, how requeuing the Node helps? I think the next attemp will still get here and requeue again?

Reply:
Hi @tnqn, I think the TODO is highlighting it doesn't check the configuration change here but return directly. it's an assumption that it's a temporary issue if last remove action failed in `removeStaleTunnelPorts`, so re-queue will help because `removeStaleTunnelPorts` also checks configuration difference. but it indeed will go back here if failed again, is there any case in your mind port can't be removed?

Reply:
My understanding was that we would call `c.ovsBridgeClient.DeletePort` and `c.interfaceStore.DeleteInterface` here, and in case of error during port removal, return an error and let the Node be requeued. `removeStaleTunnelPorts` is only called on controller start up and transient errors are always possible.

Reply:
sure, I will fix this.